src/handlers.ts

```diff
@@ -543,30 +546,103 @@ export async function postPublishedWebhook(
     });
 }
 
+export async function getSiteDataHandler(
+    ctx: Context<{ Variables: HonoContextVariables }>,
+) {
+    const request = ctx.req;
+    const host = request.header('host');
+    if (!host) {
+        // TODO handle
+        throw new Error('No Host header');
+    }
+
+    const handle = ACTOR_DEFAULT_HANDLE;
+    const apCtx = fedify.createContext(ctx.req.raw as Request, {
+        db: ctx.get('db'),
+        globaldb: ctx.get('globaldb'),
+        logger: ctx.get('logger'),
+    });
+
+    const site = await getSite(host, true);
+    await getUserData(apCtx, handle);
+    const updated = await updateSiteActor(
+        ctx.get('db'),
+        ctx.get('logger'),
+        host,
+    );
+
+    // Publish activity if the site settings have changed
+    if (updated) {
+        const actor = await apCtx.getActor(handle);
+
+        const update = new Update({
+            id: apCtx.getObjectUri(Update, { id: uuidv4() }),
+            actor: actor?.id,
+            to: PUBLIC_COLLECTION,
+            object: actor?.id,
+            cc: apCtx.getFollowersUri('index'),
+        });
+
+        await ctx
+            .get('globaldb')
+            .set([update.id!.href], await update.toJsonLd());
+
+        await apCtx.sendActivity({ handle }, 'followers', update, {
+            preferSharedInbox: true,
+        });
+    }
```

Comment:
Should this also be apart of `updateSiteActor` and reused across the 2 handlers?